app/controllers/api/v1/reservations_controller.rb

```diff
@@ -14,19 +14,28 @@ def show
       end
 
       def create
-        reservation = Reservation.new(create_reservation_params)
-        if reservation.save
-          render json: reservation, status: :created
+        if @current_user.id == params[:reservation][:user_id]
+          reservation = Reservation.new(create_reservation_params)
+
+          if reservation.save
+            render json: reservation, status: :created
+          else
+            render Json: reservation.errors.full_messages, status: :unprocessable_entity
+          end
         else
-          render Json: reservation.errors.full_messages, status: :unprocessable_entity
+          render json: { error_message: 'users can only create their own reservations' }, status: :unauthorized

```

Comment:
So basically you are implementing an Authorization system here.

Reply:
@wandergithub, Maybe, somehow, although I'm not sure. I'm just trying to prevent another user from doing what only the user himself should do 🤔